src/Maker/MakeListener.php

```diff
@@ -0,0 +1,257 @@
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\EventRegistry;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Util\UseStatementGenerator;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ * @author Steven Renaux <steven.renaux8000@gmail.com>
+ */
+final class MakeListener extends AbstractMaker
+{
+    private const ALL_TYPES = ['Listener', 'Subscriber'];
+    private bool $isSubscriber = false;
+
+    public function __construct(private readonly EventRegistry $eventRegistry)
+    {
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:listener';
+    }
+
+    /**
+     * @deprecated remove this method when removing make:subscriber
+     */
+    public static function getCommandAlias(): string
+    {
+        return 'make:subscriber';
+    }
+
+    public static function getCommandDescription(): string
+    {
+        return 'Creates a new event subscriber class or a new event listener class';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command
+            ->addArgument('name', InputArgument::OPTIONAL, 'Choose a class name for your event listener or subscriber (e.g. <fg=yellow>ExceptionListener</> or <fg=yellow>ExceptionSubscriber</>)')
+            ->addArgument('event', InputArgument::OPTIONAL, 'What event do you want to listen to?')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeEvent.txt'))
+        ;
+
+        $inputConfig->setArgumentAsNonInteractive('event');
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        /* @deprecated remove the following block when removing make:subscriber */
+        $this->handleDeprecatedMakerCommands($input, $io);
+
+        $io->writeln('');
+
+        $name = $input->getArgument('name');
+
+        if (!str_ends_with($name, 'Subscriber') && !str_ends_with($name, 'Listener')) {
+            $classToGenerate = null;
+
+            while (null === $classToGenerate) {
+                $defaultEvent = 'Listener';
+                $question = new Question('What event class to generate(enter <comment>?</comment> to see all types)', $defaultEvent);
+                $question->setAutocompleterValues(self::ALL_TYPES);
```

Comment:
This makes more sense as a choice question: either choose `Listener` or `Susbcriber`.

Reply:
Really better, thanks for the tips :wink: